### PR TITLE
Implement an ElementAt extension method

### PR DESCRIPTION
**Pull request by cocoatomo**
Following [Enumerable.ElementAt](https://msdn.microsoft.com/ja-jp/library/bb299233%28v=vs.100%29.aspx)

### CODE REVIEW

File: Funcy/FuncyList.cs [hunk 1]
```diff
@@ -316,6 +316,28 @@ public override FuncyList<TReturn> ComputeWith<TReturn>(Func<T, FuncyList<TRetur
...
+            }
+
+            if (source is Nil<T>)
+            {
+                return Maybe<T>.None();
+            }
+
+            Cons<T> cons = source.ToCons();
+
+            return FuncyListNT.ElementAt<T>(cons.Tail, index - 1);
```
Comment (Gab-km) [inline Funcy/FuncyList.cs]: Should this method be recursive? Is the following one no good?
```csharp
if (source is Nil<T>) { /* (snip) */ }

var count = source.Count();
if (index >= count)
{
    return Maybe<T>.None();
}
return Maybe<T>.Some(((IEnumerable<T>)source).ElementAt(index));
```

**Comment by cocoatomo** [comment]
Yes, you are right.
I considered a cost for object creation, but not for recursive call. Iterative process is prefer, especially Funcy inside.

File: Funcy/FuncyList.cs (hunk 1 above)
Comment (cocoatomo) [inline Funcy/FuncyList.cs]: Yes, you are right.
I considered a cost for object creation, but not for recursive call. Iterative process is prefer, especially Funcy inside.

**Comment by Gab-km** [comment]
@cocoatomo It looks simpler than the former one. :+1:

**Pull request merged**